From 669e318c2bf6660b731a205ed4d29f941085ce76 Mon Sep 17 00:00:00 2001
From: Hiren Mehta <hiren@pensando.io>
Date: Thu, 16 Feb 2023 22:15:22 -0800
Subject: [PATCH 25/30] arm64: dts: pensando: psci support

Change CPU enable-method from 'spin-table' to 'psci'.
Update flash obfl partition.

Signed-off-by: Hiren Mehta <hiren@pensando.io>
Signed-off-by: Ashwin H <Ashwin.H@amd.com>
---
 arch/arm64/boot/dts/pensando/Makefile         |  3 +-
 .../boot/dts/pensando/elba-16core-psci.dtsi   | 57 ++++++++++++++++
 .../boot/dts/pensando/elba-asic-psci.dts      | 26 +++++++
 arch/arm64/boot/dts/pensando/elba-asic.dts    |  8 ---
 .../boot/dts/pensando/elba-flash-parts.dtsi   |  5 ++
 arch/arm64/boot/dts/pensando/elba.dtsi        | 68 +++++++++++++++----
 6 files changed, 144 insertions(+), 23 deletions(-)
 create mode 100644 arch/arm64/boot/dts/pensando/elba-16core-psci.dtsi
 create mode 100644 arch/arm64/boot/dts/pensando/elba-asic-psci.dts

diff --git a/arch/arm64/boot/dts/pensando/Makefile b/arch/arm64/boot/dts/pensando/Makefile
index 8971be416..a1ade7e98 100644
--- a/arch/arm64/boot/dts/pensando/Makefile
+++ b/arch/arm64/boot/dts/pensando/Makefile
@@ -1,6 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
-dtb-$(CONFIG_ARCH_PENSANDO_ELBA_SOC) += \
-		elba-asic.dtb
+dtb-$(CONFIG_ARCH_PENSANDO_ELBA_SOC) += elba-asic-psci.dtb
 
 always-y	:= $(dtb-y)
 subdir-y	:= $(dts-dirs)
diff --git a/arch/arm64/boot/dts/pensando/elba-16core-psci.dtsi b/arch/arm64/boot/dts/pensando/elba-16core-psci.dtsi
new file mode 100644
index 000000000..099bfaec8
--- /dev/null
+++ b/arch/arm64/boot/dts/pensando/elba-16core-psci.dtsi
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-2.0
+
+&cpu0 {
+	enable-method = "psci";
+};
+&cpu1 {
+	enable-method = "psci";
+};
+&cpu2 {
+	enable-method = "psci";
+};
+&cpu3 {
+	enable-method = "psci";
+};
+&cpu4 {
+	enable-method = "psci";
+};
+&cpu5 {
+	enable-method = "psci";
+};
+&cpu6 {
+	enable-method = "psci";
+};
+&cpu7 {
+	enable-method = "psci";
+};
+&cpu8 {
+	enable-method = "psci";
+};
+&cpu9 {
+	enable-method = "psci";
+};
+&cpu10 {
+	enable-method = "psci";
+};
+&cpu11 {
+	enable-method = "psci";
+};
+&cpu12 {
+	enable-method = "psci";
+};
+&cpu13 {
+	enable-method = "psci";
+};
+&cpu14 {
+	enable-method = "psci";
+};
+&cpu15 {
+	enable-method = "psci";
+};
+
+/ {
+	psci {
+		compatible = "arm,psci-0.2";
+		method = "smc";
+	};
+};
diff --git a/arch/arm64/boot/dts/pensando/elba-asic-psci.dts b/arch/arm64/boot/dts/pensando/elba-asic-psci.dts
new file mode 100644
index 000000000..dfa6deaa0
--- /dev/null
+++ b/arch/arm64/boot/dts/pensando/elba-asic-psci.dts
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: GPL-2.0
+
+/dts-v1/;
+
+/ {
+	model = "Elba ASIC Board";
+	compatible = "pensando,elba";
+
+	aliases {
+		serial0 = &uart0;
+		spi0 = &spi0;
+		spi1 = &qspi;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+#include "elba.dtsi"
+#include "elba-16core.dtsi"
+#include "elba-16core-psci.dtsi"
+#include "elba-asic-common.dtsi"
+#include "elba-flash-parts.dtsi"
+#include "elba-asic-memory.dtsi"
+#include "elba-mdev.dtsi"
diff --git a/arch/arm64/boot/dts/pensando/elba-asic.dts b/arch/arm64/boot/dts/pensando/elba-asic.dts
index a1ba7b245..70d0dbd38 100644
--- a/arch/arm64/boot/dts/pensando/elba-asic.dts
+++ b/arch/arm64/boot/dts/pensando/elba-asic.dts
@@ -24,11 +24,3 @@
 #include "elba-flash-parts.dtsi"
 #include "elba-asic-memory.dtsi"
 #include "elba-mdev.dtsi"
-
-/ {
-       firmware {
-               penfw {
-                       compatible = "pensando,penfw";
-               };
-       };
-};
diff --git a/arch/arm64/boot/dts/pensando/elba-flash-parts.dtsi b/arch/arm64/boot/dts/pensando/elba-flash-parts.dtsi
index 2b5dc16df..4ce934081 100644
--- a/arch/arm64/boot/dts/pensando/elba-flash-parts.dtsi
+++ b/arch/arm64/boot/dts/pensando/elba-flash-parts.dtsi
@@ -90,6 +90,11 @@
 			reg = <0x6a00000 0x400000>;
 		};
 
+		partition@6e00000 {
+			label = "flashobfl";
+			reg = <0x6e00000 0x1200000>;
+		};
+
 		partition@8000000 {
 			label = "diagfw";
 			reg = <0x8000000 0x7fe0000>;
diff --git a/arch/arm64/boot/dts/pensando/elba.dtsi b/arch/arm64/boot/dts/pensando/elba.dtsi
index 8354aff97..b68a3aebc 100644
--- a/arch/arm64/boot/dts/pensando/elba.dtsi
+++ b/arch/arm64/boot/dts/pensando/elba.dtsi
@@ -31,6 +31,11 @@
 		#clock-cells = <0>;
 	};
 
+	gmac_clk: oscillator5 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+	};
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupts = <GIC_PPI 13 (GIC_CPU_MASK_SIMPLE(1) |
@@ -152,7 +157,7 @@
 				reg = <0>;
 				gpio-controller;
 				#gpio-cells = <2>;
-				snps,nr-gpios = <8>;
+				ngpios = <8>;
 				interrupts = <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
 				interrupt-controller;
 				interrupt-parent = <&gic>;
@@ -164,7 +169,7 @@
 				reg = <1>;
 				gpio-controller;
 				#gpio-cells = <2>;
-				snps,nr-gpios = <8>;
+				ngpios = <8>;
 			};
 		};
 
@@ -206,17 +211,6 @@
 			};
 		};
 
-		kpcimgr@20000000 {
-			compatible = "pensando,kpcimgr";
-			msi-parent = <&its 0xf>;
-			shmem-index = <3>;
-			hwmem-index = <2>;
-			reg =  <0x0 0x20000000 0x0 0x00380000   /* PXB Base */
-				0x0 0x61800000 0x0 0x00101000   /* Interrupts */
-				0x0 0xc4000000 0x0 0x01000000   /* HWMEM */
-				0x0 0x0        0x0 0x00f00000>; /* SHMEM */
-		};
-
 		/*
 		 * Until we  know the interrupt domain following this, we
 		 * are forced to use this is the place where interrupts from
@@ -249,6 +243,33 @@
 			enable_mask = <(1 << 0)>;
 		};
 
+		stmmac_axi_setup: stmmac-axi-config {
+			snps,wr_osr_lmt = <0xf>;
+			snps,rd_osr_lmt = <0xf>;
+			snps,blen = <256 128 64 32 0 0 0>;
+		};
+
+		gmac0: ethernet@6f21a000 {
+			compatible = "snps,dwmac-3.70a", "snps,dwmac";
+			reg = <0x0 0x6f21a000 0x0 0x8000>;
+			clocks = <&gmac_clk>;
+			clock-names = "stmmaceth";
+			interrupts = <GIC_SPI 29 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "macirq";
+			mac-address = [00 00 00 00 00 00];/* Filled in by U-Boot */
+			max-frame-size = <3800>;
+			snps,multicast-filter-bins = <256>;
+			snps,perfect-filter-entries = <128>;
+			rx-fifo-depth = <16384>;
+			tx-fifo-depth = <16384>;
+			snps,axi-config = <&stmmac_axi_setup>;
+			snps,pbl = <2>;
+			fixed-link {
+				speed = <100>;
+				full-duplex;
+			};
+		};
+
 		emmc: mmc@30440000 {
 			compatible = "pensando,elba-emmc", "cdns,sd4hc";
 			clocks = <&emmc_clk>;
@@ -259,6 +280,8 @@
 			cdns,phy-input-delay-legacy = <0x4>;
 			cdns,phy-input-delay-sd-uhs-sdr50 = <0x6>;
 			cdns,phy-input-delay-sd-uhs-ddr50 = <0x16>;
+			cdns,phy-dll-delay-sdclk = <0>;
+			cdns,phy-dll-delay-sdclk-hsmmc = <0>;
 			cdns,mmc-ddr-1_8v;
 			status = "disabled";
 		};
@@ -279,6 +302,25 @@
 			       0x0 0x1400 0x0 0x10		/* WDT0 */
 			       0x0 0x20000000 0x0 0x380000>;	/* PXB Base */
 		};
+
+		kpcimgr@20000000 {
+			compatible = "pensando,kpcimgr";
+			msi-parent = <&its 0xf>;
+			shmem-index = <3>;
+			hwmem-index = <2>;
+			reg =  <0x0 0x20000000 0x0 0x00380000   /* PXB Base */
+				0x0 0x61800000 0x0 0x00101000   /* Interrupts */
+				0x0 0xc4000000 0x0 0x01000000   /* HWMEM */
+				0x0 0x0        0x0 0x00F00000>; /* SHMEM */
+		};
+
+		/*
+		 * panic: panicdump@0 {
+		 *	compatible = "pensando,capri-crash";
+		 *	status = "disabled";
+		 * };
+		 */
+
 		sbus@307c24a8 {
 			compatible = "pensando,sbus";
 			sbus-ring-num = <0>;
-- 
2.17.1

